ros/src/tl_detector/light_classification/tl_classifier_cnn.py

```diff
@@ -0,0 +1,583 @@
+# -------------------------------------------------------------------------------
+# Author: Alexey Simonov <alexey.simonov@gmail.com>
+# Date:   28.08.17
+# -------------------------------------------------------------------------------
+
+"""Traffic Lights Classifier using CNN
+
+TLClassifierCNN implements a multi-class classifier for traffic lights images.
+Using convolutional neural net architecture inspired by CIFAR-10.
+Pure tensorflow implementation.
+
+TLLabelConverter is a helper class to convert from string label representation
+to numeric representation for internal handling/training in tensorflow.
+
+Developed with tensorflow 1.1 (GPU) on python 2.7.13 (conda).
+
+Takes an image of size 32x32 with 3 channels in OpenCV convention (BGR uint8)
+
+Trained on Bosch Small Traffic Lights dataset.
+
+Todo:
+    * improve training as it seems to stop improving after handful of epochs.
+      may need truncation of sofmax layer. requires analysis in tensorboard
+    * for training balance the number of examples in different classes. may
+      need augmentation
+    * move to generator-type inputs
+    * use tensorflow Dataset interface
+    * use tensorflow Estimator interface?
+    * add embeddings visualisation to tensorboard
+"""
+
+import os
+import math
+from datetime import datetime
+
+import numpy as np
+import tensorflow as tf
+try:
+    # only used in train() method. when run on ROS train() should never be called
+    from tqdm import tqdm
```

Comment:
I understand why you did it, but I don't like that. I guess we need to decide whether the code in the Panda repository needs the training part of the code. In principle it will never be used. Let me know what's your view on that.

Reply:
I think we should include it for our own future reference. In a real project yes, it should be a separate mini project.

On the other hand, we can create more projects in the panda organization and include the training part of the projects there.

what do you think?

Reply:
What I meant is that we should not include the experiments here. If it comes to the final work, by all means! But I would put this code in a separate repo within the organization. For my part, I will do a blog post and publish the whole thing anyways because it's not specific to any dataset - it can work with anything.